service/history/history_engine.go

```diff
@@ -1041,8 +1036,15 @@ func (e *historyEngineImpl) ListTasks(
 }
 
 // StateMachineEnvironment implements shard.Engine.
-func (e *historyEngineImpl) StateMachineEnvironment() hsm.Environment {
-	return e.stateMachineEnvironment
+func (e *historyEngineImpl) StateMachineEnvironment(
+	operationTag metrics.Tag,
+) hsm.Environment {
+	return &stateMachineEnvironment{
```

Comment:
I assume changing from singleton to "always new" is ok here.

Reply:
Yup. No state stored in stateMachineEnvironment.